### ISSUE TITLE
Polish the signature of CallApplicationMain in klr

### ISSUE THREAD

**Issue by troydai** [aspnet/dnx#1043]
Here's the function pointer declaration (https://github.com/aspnet/KRuntime/blob/dev/src/klr/klr.h#L11)

``` C++
typedef struct CALL_APPLICATION_MAIN_DATA
{
    LPCWSTR applicationBase; // application base of managed domain
    LPCWSTR klrDirectory; // path to KLR helper directory
    int argc; // Number of args in argv
    LPCWSTR* argv; // Array of arguments
    int exitcode; // Exit code from Managed Application
} *PCALL_APPLICATION_MAIN_DATA;

typedef HRESULT (STDAPICALLTYPE *FnCallApplicationMain)(
    PCALL_APPLICATION_MAIN_DATA pCallApplicationMainData
    );
```

The `pCallApplicationMainData` will contain the exit code, there is no need to return the it from the method again. The return type can be instead change to `bool` so that the caller can be further simplified.

**Comment by troydai** [comment]
@glennc @muratg If we're not going to fix it, should we just close it?

**Comment by muratg** [comment]
@davidfowl @lodejard What are your thoughts on this one?

**Comment by davidfowl** [comment]
Part of any clean up that @moozzyk does

**Comment by moozzyk** [comment]
The `HRESULT` returned by the `FnCallApplicationMain` does not represent the return value of the target function but the result of the operation of calling the target function (e.g. if for whatever reason we cannot start CoreClr the FnCallApplicationMain will return the HRESULT representing the failure and will not set the `exitCode`). 

What we should probably fix is actually to remove this line:
https://github.com/aspnet/dnx/blob/dev/src/dnx.clr/dnx.clr.cpp#L30

**Comment by moozzyk** [comment]
Fixed in #2857
